ui/src/ReactCustomHooks/useTranslate/useTranslate.hook.js

```diff
@@ -0,0 +1,46 @@
+/* eslint-disable react/display-name */
+import React, { useMemo, Fragment } from 'react';
+import { merge } from 'lodash-es';
+import commonTranslations from './common.i18n.json';
+
+const DEFAULT_LOCALE = 'en';
+
+const STRIP_INSERT_REGEX = /\${|}/;
+
+export const useTranslate = (
+  bundle = {},
+  currentLocale = navigator.language
+) => {
+  return useMemo(() => {
+    // calculate the new locale, and get the bundle for it
+    const allTranslationBundles = merge({}, commonTranslations, bundle);
+
+    const locale = currentLocale.split(/_|-/)[0];
+
+    const translationsForLocale = allTranslationBundles[locale]
+      ? allTranslationBundles[locale]
+      : allTranslationBundles[DEFAULT_LOCALE];
+
+    return (key, insertsToProcess = {}) => {
+      const valueForKey = translationsForLocale[key]
+        ? translationsForLocale[key]
+        : `${key} provided, but none found for locale ${locale}`;
+
+      // look at the string, and split it on the insert delimiters
+      // for a string 'foo ${insert} bar', output will be ['foo ', 'insert', ' bar']
+      // check each entry if it matches an insert key. If so, use the insert value,
+      // else, use the value from the split
+      const parsedValue = valueForKey
+        .split(STRIP_INSERT_REGEX)
+        .reduce(
+          (acc, part) =>
+            insertsToProcess[part]
+              ? acc.concat([insertsToProcess[part]])
+              : acc.concat([part]),
+          []
+        );
+
+      return <Fragment>{parsedValue}</Fragment>;
```

Comment:
We'll need a non-fragment version for translating into strings (aria attributes and other labels)

Reply:
I'm curious if we will - from a DOM pov, the fragment doesnt exist, meaning we just have text. What is the case you are thinking of?

Reply:
Case was calling translate to provide a HTML attribute value. This currently returns [object Object]. Will add a 'asString' option for these cases